frontend/src/pages/ProgressionChecker/ProgressionChecker.jsx

```diff
@@ -42,29 +43,34 @@ const ProgressionChecker = () => {
   const [isLoading, setIsLoading] = useState(true);
 
   const {
-    programCode, specialisation, minor,
+    programCode, majors, minor,
   } = useSelector((state) => state.degree);
 
   const [structure, setStructure] = useState({});
 
   useEffect(() => {
     // get structure of degree
     const fetchStructure = async () => {
-      try {
-        const res = await axios.get(`/programs/getStructure/${programCode}/${specialisation}${minor && `/${minor}`}`);
-        setStructure(res.data.structure);
-      } catch (err) {
-        // eslint-disable-next-line no-console
-        console.log(err);
-      }
+      await Promise.all(majors.map(async (major) => {
+        try {
+          const newStructure = structure;
+          const res = await axios.get(`/programs/getStructure/${programCode}/${major}${minor && `/${minor}`}`);
+          newStructure[major] = res.data.structure;
+          setStructure(newStructure);
+        } catch (err) {
+          // eslint-disable-next-line no-console
+          console.log(err);
+        }
+      }));
       setIsLoading(false);
     };
-    if (programCode && specialisation) fetchStructure();
-  }, [programCode, specialisation, minor]);
+    if (programCode && majors) fetchStructure();
```

Comment:
May want to check `majors.length` instead since `majors` will always be initialised to an empty array (i.e. since `[]` is eval to true in js

Reply:
oh right, I see thanks will check that length > 0